plugin-gradle/src/main/java/com/diffplug/gradle/spotless/TypescriptExtension.java

```diff
@@ -54,6 +72,18 @@ public TypescriptFormatExtension tsfmt() {
 		@Nullable
 		Object configFilePath = null;
 
+		private final Map<String, String> versions = new TreeMap<>();
+
+		public TypescriptFormatExtension() {
+			this(DEFAULT_TYPESCRIPT_FORMATTER_VERSION, DEFAULT_TYPESCRIPT_FORMATTER_TYPESCRIPT_VERSION, DEFAULT_TYPESCRIPT_FORMATTER_TSLINT_VERSION);
+		}
+
+		public TypescriptFormatExtension(String formatterVersion, String typescriptVersion, String tslintVersion) {
+			this.versions.put("formatterVersion", requireNonNull(formatterVersion));
+			this.versions.put("typescriptVersion", requireNonNull(typescriptVersion));
+			this.versions.put("tslintVersion", requireNonNull(tslintVersion));
```

Comment:
I don't feel strongly about this, but it might make sense for these to be `public static final String NPM_PKG_TSLINT`, etc. defined in the step.  I'm fine with this as-is as well.

Reply:
you are talking about the map-keys, correct?

Reply:
correct